cpp/.clang-format

```diff
@@ -0,0 +1,30 @@
+---
+Language: Cpp
+Standard: Cpp11
```

Comment:
Just realized we can probably change this to Cpp17, or whatever we decide to require...maybe our customers will need support back as far as 14?